syntax = "proto3";

import "map_renderer.proto";
import "transport_router.proto";
//import "graph.proto";

package tc_serialize;

message Coordinates {
  double longitude = 1;
  double latitude = 2;
}

message Stop {
  bytes stop_name = 1;
  Coordinates coordinates = 2;
}

message Bus {
  bytes bus_name = 1;
  repeated bytes stop_in_bus_route = 2;
  bool is_circle = 3;
}

message DistanceBetweenStop {
  bytes stop_from_name = 1;
  bytes stop_to_name = 2;
  double distance = 3;
}

message TransportCatalogueCore {
  repeated Stop stops = 1;
  repeated Bus buses = 2;
  repeated DistanceBetweenStop distance_between_stop = 3;
}


message TC {
  TransportCatalogueCore transport_cat_core = 1;
  map_proto.Map map_core = 2;
  router_proto.TransportRouter router_core = 3;
}